internal/config/identity/ldap/ldap.go

```diff
@@ -329,3 +330,61 @@ func (l *Config) LookupGroupMemberships(userDistNames []string, userDNToUsername
 
 	return res, nil
 }
+
+// LookupUsername searches for the username and groups of a given DN, like opposite of LookupUserDN
+func (l *Config) LookupUsername(userDN string) (string, []string, error) {
```

Comment:
I don't think there's a robust way to do this. We can probably stick with only recording the DN with the service account if the user provided a DN in the API input. If they provide a short username, we lookup the DN and record both. We can remove this function.

In any situation where the short username is given, we first get the full DN and use that for the operation.